### ISSUE TITLE
Text type definition is unclear

### ISSUE THREAD

**Issue by MatthiasWiesmann** [schemaorg/schemaorg#3033]
The definition for the text data-type (https://schema.org/Text) is not well defined.
The assumption seems to be that it contains Unicode text, but this is not stated explicitly and leaves many ambiguities: 

* What characters are allowed? Are C0 and C1 control characters allowed? 
* How are white-space characters handled? The current validator just echoes back text fields, but this this collapses all white-spaces to a single space (0x20). De facto this implies that carriage returns and paragraph cannot exist in text-fields. 
* Is the content expected to be raw text? Are some tagged formats allowed (HTML, markdown), or extensions (ANSI escape sequences).

What is value of `description` of the following snippets: 

```
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "Product",
  "description": "Line 1\nLine 2"
}
</script>
```

```
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "Product",
  "description": "Bell\u0007Pepper"
}
</script>
```

If the following snippet is allowed:

```
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "Product",
  "description": "Line 1<br>Line 2"
}
</script>
```

Is it equivalent to the following? 

```
<div itemscope itemtype="https://schema.org/Product">
  <span itemprop="description">Line 1<br>Line 2</span>
</div>
```

**Comment by jakubklimek** [comment]
In addition, what is the relation of the `schema:Text` datatype to the standard RDF `rdf:langString` with language tags determining the language of the text?

**Comment by HughP** [comment]
@jakubklimek I'm not aware of any value "modifier" or attribute directly indicating language in any of schema's key:value pairs. However, I imagine that one could construct an array such that one could use https://schema.org/inLanguage and in that context BCP47 codes are used.

**Comment by jakubklimek** [comment]
@HughP Yes, exactly, it is not simple to do this using `schema:Text`. That is why I am wondering, why not to use the standard RDF `rdf:langString` instead of `schema:Text`.
I.e.
```turtle
[] schema:name "My name"@en, "Moje jméno"@cs .
```
instead of 
```turtle
[] schema:name "My name"^^schema:Text, "Moje jméno"^^schema:Text
```
and then thinking about a non-standard way of indicating which one of those is in English and which one is in Czech...

**Comment by RichardWallis** [comment]
The basic Schema Text datatype defines no conditions as to what is contained within the associated string of characters.

What is possible for a string is influenced by the environment and markup syntax in which the data is presented.

In browsers this usually means UTF-8.

As to language tagging that depends on the serialisation used JSON-LD defines one approach, RDFa another.  

Check out this example https://schema.org/isAccessibleForFree#eg-0437 to see both.

**Comment by MatthiasWiesmann** [comment]
I think this should be made explicit in the documentation, in particular that the behaviour/semantics of control characters like carriage return or tab is _undefined_. 

This also means that the `Text` type is unsuitable for text beyond a paragraph, and IMHO an argument in favour of supporting creative work for descriptions (#2942).

**Comment by RichardWallis** [comment]
I agree, some indication of what a Text value can contain would be helpful.  More importantly what constraints are _**not**_ applied to its content.

It is closely allied to the rdf:langstring ie. A Unicode [[UNICODE](https://www.w3.org/TR/rdf11-concepts/#bib-UNICODE)] string, with an optional language-tag.

How the contents of a Text value are serialised including escaping of characters with specific functionality, and any supplied language-tage, is dependant on the serialisation method chosen (Microdata, RDFa, JASON-LD).

Interpretation of control characters and any complex formatting such as HTML, Markdown, etc. are not defined and open to interpretation by data consumers.

**Comment by MatthiasWiesmann** [comment]
I somehow feel that pushing the problem to consumers just means a lot of cases are undefined.  

* Text that defines Unicode control characters is undefined. 
* Text that contain `<> &` is undefined (HTML)
* Text that contains `#_*` is undefined (Markdown)
* Text that contains `{}\` is undefined (Latex, RTF)

**Comment by RichardWallis** [comment]
That is the nature of an undefined Text field , a bit like the rdf:langstring, it contains Unicode characters.

If you wish to constrain the use of this generic unicode character container in any way there are options within the types it is used (eg. [encodingFormat](https://schema.org/encodingFormat) in CreativeWork.  Or potentially there is a case for proposing a formatted text subtype of Text that could indicate the type of expected text format, Or potentially there could be a proposal for a HTMLText, LatexText, etc. subtypes.

In the meantime we should be more clear as to the contents of a Text being undefined as to formatting.
